pymeasure/instruments/instrument.py

```diff
@@ -59,13 +59,16 @@ class Instrument(CommonBase):
     :param adapter: A string, integer, or :py:class:`~pymeasure.adapters.Adapter` subclass object
     :param string name: The name of the instrument. Often the model designation by default.
     :param includeSCPI: A boolean, which toggles the inclusion of standard SCPI commands
+    :param setup: A method executed between instantiation and creation of children (channels).
+    :param setup_kwargs: Keyword arguments for the setup method.
```

Comment:
You could mandate that `setup` is a callable without arguments. Then there's one less argument in the signature, and people needing such an advanced pattern can use `partial(somefunc, arg1, arg2)` or a lambda to define such a callable.

Reply:
Either that, or we tell the people to use add_child if they require instrument communication before channel communication (see main discussion).

Reply:
I remove it for now.